Improve :has() pseudo-class documentation (#20889)
mdn/content: files/en-us/web/css/css_selectors/index.md

    **Syntax:** `A + B`

    **Example:** `h2 + p` will match the first {{HTMLElement("p")}} element that _immediately_ follow an {{HTMLElement("h2")}} element.

- [Column combinator](/en-US/docs/Web/CSS/Column_combinator) {{Experimental_Inline}}
  - : The `||` combinator selects nodes which belong to a column.

    **Syntax:** `A || B`

    **Example:** `col || td` will match all {{HTMLElement("td")}} elements that belong to the scope of the {{HTMLElement("col")}}.

## Pseudo-classes and pseudo-elements

- [Pseudo classes](/en-US/docs/Web/CSS/Pseudo-classes)

  - : The `:` pseudo allow the selection of elements based on state information that is not contained in the document tree.

    **Example:** `a:visited` will match all {{HTMLElement("a")}} elements that have been visited by the user.

- [Pseudo elements](/en-US/docs/Web/CSS/Pseudo-elements)
  - : The `::` pseudo represent entities that are not included in HTML.

    **Example:** `p::first-line` will match the first line of all {{HTMLElement("p")}} elements.

## Structure of a selector

The term 'selector' can refer to one of the following:

- Simple selector

>   - : An individual selector. A simple selector represents a single condition on an element. A given element is said to match a simple selector when that simple selector accurately describes the element. All [basic selectors](#basic_selectors) and [pseudo-classes](#pseudo-classes_and_pseudo-elements) are simple selectors.

Reviewer: ```suggestion
  - : A selector with a single component such as a single id selector, type selector, classesselector, including a single pseudo-class, attribute, or pseudo-element, not used in combination with or containing any other selector component or combinator.  
```
we need to bring all of these descriptions to a 3rd to 8th grade level. This sounds spec-y (same for compound and complex)

  
    A [type selector](/en-US/docs/Web/CSS/Type_selectors) or a [universal selector](/en-US/docs/Web/CSS/Universal_selectors) in a compound selector must come first in the sequence of selectors. A pseudo-class must follow the type selector or universal selector if present. Only one type selector or a universal selector is allowed in the sequence. Since whitespace represents the [descendant combinator](/en-US/docs/Web/CSS/Descendant_combinator), no whitespace is allowed between the simple selectors in a compound selector.

    **Example:** `a#selected {...}`

- Complex selector
  - : A sequence of one or more [compound selectors](/en-US/docs/Web/CSS/CSS_Selectors#compound_selector) that are separated by [combinators](#combinators). A complex selector represents a set of simultaneous conditions on a set of elements. These conditions apply in the context of relationships described by the combinators. A given element is said to match a complex selector when the element matches compound selectors and the combinators between the compound selectors.

    **Example:** `a#selected > .icon {...}`, `.box h2 + p {...}`

- Relative selector

>   - : A complex selector beginning with a combinator. The [descendant combinator](/en-US/docs/Web/CSS/Descendant_combinator) is implied in case of a missing combinator.

Reviewer: ```suggestion
  - : A selector that represents an element relative to one or more anchor elements preceded by a combinator. Relative selectors that don't begin with an explicit [combinator](#combinators) have an implied [descendant combinator](/en-US/docs/Web/CSS/Descendant_combinator) assumed.
```

The spec does not explain anchor. Asking the author.... my edit likely isn't an improvement, but maybe.

Author: I added a line about anchor elements in the pseudo-classes page

    **Example:** `+ div#topic > #reference {...}`

- [Selector list](/en-US/docs/Web/CSS/Selector_list)

>   - : A comma-separated list of [simple](/en-US/docs/Web/CSS/CSS_Selectors#simple_selector), [compound](/en-US/docs/Web/CSS/CSS_Selectors#compound_selector), or [complex](/en-US/docs/Web/CSS/CSS_Selectors#complex_selector) selectors. If the constituent selector type of a selector list is important but unspecified, it is called a _complex selector list_. A given element is said to match a selector list when the element matches any (at least one) of the selectors in that selector list. Read more about when a selector list is deemed [invalid](/en-US/docs/Web/CSS/Selector_list#invalid_selector_list) and how to construct a [forgiving selector list](/en-US/docs/Web/CSS/Selector_list#forgiving_selector_list).

Author: Spec wording. Would you know how to reword it to make it less ambiguous?

- [Pseudo elements](/en-US/docs/Web/CSS/Pseudo-elements)
  - : The `::` pseudo represent entities that are not included in HTML.

    **Example:** `p::first-line` will match the first line of all {{HTMLElement("p")}} elements.

## Structure of a selector

The term 'selector' can refer to one of the following:

- Simple selector
  - : A selector with a single component, such as a single id selector or type selector, that's not used in combination with or contains any other selector component or combinator. A given element is said to match a simple selector when that simple selector accurately describes the element. All [basic selectors](#basic_selectors), attributes, and single [pseudo-classes and pseudo-elements](#pseudo-classes_and_pseudo-elements) are simple selectors.

- Compound selector
  - : A sequence of [simple selectors](#simple_selector) that are not separated by a [combinator](#combinators). A compound selector represents a set of simultaneous conditions on a single element. A given element is said to match a compound selector when the element matches all the simple selectors in the compound selector.

>     In a compound selector, a [type selector](/en-US/docs/Web/CSS/Type_selectors) or a [universal selector](/en-US/docs/Web/CSS/Universal_selectors) in a compound selector must come first in the sequence of selectors. Only one type selector or a universal selector is allowed in the sequence. Since whitespace represents the [descendant combinator](/en-US/docs/Web/CSS/Descendant_combinator), no whitespace is allowed between the simple selectors in a compound selector.

Reviewer: ```suggestion
    In a compound selector, the [type selector](/en-US/docs/Web/CSS/Type_selectors) or [universal selector](/en-US/docs/Web/CSS/Universal_selectors) must come first in the sequence of selectors. Only one type selector or the universal selector is allowed in the sequence. Since whitespace represents the [descendant combinator](/en-US/docs/Web/CSS/Descendant_combinator), no whitespace is allowed between the simple selectors in a compound selector.
```
